### ISSUE TITLE
Mongodb pod fails to use existing PVC during an upgrade

### ISSUE THREAD

**Issue by windowsrefund** [kubeshop/helm-charts#842]
I just deployed 1.17.46 as an upgrade over 1.17.31 and am seeing the following issue with the testkube-mongodb pod

```
Events:
  Type     Reason              Age                 From                     Message
  ----     ------              ----                ----                     -------
  Normal   Scheduled           37m                 default-scheduler        Successfully assigned testkube/testkube-mongodb-6496746d9d-v8w7w to i-08796528506d9452d
  Warning  FailedAttachVolume  37m                 attachdetach-controller  Multi-Attach error for volume "pvc-58c4dda9-0691-4dc4-ad42-1e535e65d930" Volume is already used by pod(s) testkube-mongodb-6f6668cc54-s42hx
  Warning  FailedMount         25m                 kubelet                  Unable to attach or mount volumes: unmounted volumes=[datadir], unattached volumes=[common-scripts kube-api-access-hd9gd datadir]: timed out waiting for the condition
```

The new pod is stuck in `ContainerCreating` status.

**Comment by windowsrefund** [comment]
bumping this

**Comment by ypoplavs** [comment]
Hi @windowsrefund!
Apologies for the delay. 

The error is caused by `ReadWriteOnce` access mode of a Mongo volume. It means that the volume can be mounted as read-write by a single node. It can still allow multiple pods to access the volume but they need to reside on the same node. More about access modes: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes

There were no changes from our side to Mongo values so the pod was recreated for a different reason (perhaps it needed to be moved to another node as the current one was lacking resources or sth ) and while the volume was still used by a previous pod a new one could not attach to it. 

To avoid such issues in future you may consider using `Recreate` strategy for Deployments: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#recreate-deployment
It will guarantee termination of an older pod before a new one is launched.

**Comment by windowsrefund** [comment]
Thanks for this. I was able to upgrade as expected after adding the following into my values.yaml

```
mongodb:
  updateStrategy:
    type: Recreate
```

However, now I appear to have hit another issue. A test suite which ran fine on the previous deployment is now failing despite the fact I haven't changed anything related to minio. Here's the appropriate key of the corresponding tse

```
executionresult:
        status: failed
        output: ""
        outputtype: ""
        errormessage: 'can''t create new test execution, can''t insert into storage:
          could not create bucket: bucket "testkube-logs" already exists'
        steps: []
        reports: null
```

I'll post an update should I find the root cause

**Comment by windowsrefund** [comment]
OK so interestingly enough, the failure occurs only once post upgrade and subsequent runs pass. I'll probably close this out soon (unless my experience changes) as the primary issue has been resolved. That said, feel free to ping me if I can provide any info that may be  helpful.
